### PR TITLE
adding matchSelector and tweaked delegation pseudo

### PR DESCRIPTION
**Pull request by csuwildcat**
adding matchSelector and tweaked delegation pseudo

### CODE REVIEW

**Comment by pennyfx** [comment]
Can we get a test for the matchesSelector?

File: x-tag.js [hunk 1]
```diff
@@ -4,6 +4,7 @@
     win = window,
     head = doc.getElementsByTagName('head')[0];
 
+
```
Comment (pennyfx) [inline x-tag.js:7]: why the extra line?

File: x-tag.js [hunk 2]
```diff
@@ -148,11 +152,11 @@
...
-          var target = xtag.query(this, pseudo.value).filter(function(node){
-            return node == args[0].target || 
-              node.contains ? node.contains(args[0].target) : false;
-          })[0];
-          return target ? fn.apply(target, args) : false;
+		  var target = args[0].target,
+		      delegate = xtag.query(this.parentNode || this, pseudo.value).filter(function(node){
+	            return node == target || node.contains ? node.contains(target) : false;
+	          })[0];
+          return delegate ? fn.apply(delegate, args) : false;
```
Comment (pennyfx) [inline x-tag.js:159]: What was the purpose of this change?

Comment (csuwildcat) [inline x-tag.js:159]: Both the second added part is right, but I see an optimization for it, it
should read: var delegate = this == target ? this : xtag.filter(...);

We can change it tomorrow.

**Comment by pennyfx** [comment]
I manually added matchSelector so I'm closing this pull request.

**Pull request closed without merging**